std/typecons.d

```diff
@@ -4865,12 +4865,154 @@ unittest
     assert(rc1 == 5);
     // No more allocation, add just one extra reference count
     auto rc2 = rc1;
+    assert(rc1.refCountedStore.refCount == 2);
     // Reference semantics
     rc2 = 42;
     assert(rc1 == 42);
     // the pair will be freed when rc1 and rc2 go out of scope
 }
 
+// recursively destroy a class instance
+private void _destroy(T)(T obj) if (is(T == class))
```

Comment:
Eek, I thought I dealt with all attribute inference cases :(

I haven't looked into rt_finalize, but as I understand it, it does destruction as well as zero-bombing the whole class instance, crucially including the vptr. Does it do anything else? Otherwise, it could be implemented generically like `destructRecurse` did for structs and fixed-length arrays.